### PR TITLE
nrf9160: aws_fota: Fix going online only if it has gone offline

### PR DESCRIPTION
**Pull request by sigvartmh**
Fixes so that the `aws_fota` sample does not tries to go online even
if it never went offline.

Signed-off-by: Sigvart Hovland <sigvart.hovland@nordicsemi.no>

### CODE REVIEW

File: samples/nrf9160/aws_fota/src/main.c [hunk 1]
```diff
@@ -488,13 +491,17 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
 		if (err) {
 			printk("Error turning off the LTE link\n");
 		}
+		link_offline = true;
```
Comment (jtguggedal) [inline samples/nrf9160/aws_fota/src/main.c:494]: I suggest adding a `break` after line 492, as `link_offline` can be set to true now even if `lte_lc_offline` failed

Comment (sigvartmh) [reply samples/nrf9160/aws_fota/src/main.c:494]: ah yes thanks.

**Comment by jtguggedal** [comment]
Looks like there was a mixup, so the wrong commit was pushed to this branch

**Comment by sigvartmh** [comment]
Ah that's where it went :dagger:

**Comment by sigvartmh** [comment]
Now it should be back to normal @jtguggedal

File: samples/nrf9160/aws_fota/src/main.c [hunk 2]
```diff
@@ -490,14 +493,21 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
...
 		break;
 
 	case AWS_FOTA_EVT_ERASE_DONE:
-		printk("AWS_FOTA_EVT_ERASE_DONE, reconnecting the LTE link\n");
-		err = lte_lc_connect();
-		if (err) {
-			printk("Error reconnecting the LTE link\n");
+		printk("AWS_FOTA_EVT_ERASE_DONE\n");
+		if (link_offline) {
```
Comment (jtguggedal) [inline samples/nrf9160/aws_fota/src/main.c:503]: Might add blank line before this line

File: samples/nrf9160/aws_fota/src/main.c [hunk 3]
```diff
@@ -490,14 +493,21 @@ static void aws_fota_cb_handler(struct aws_fota_event *fota_evt)
...
+			printk("Reconnecting the LTE link\n");
+			err = lte_lc_connect();
```
Comment (jtguggedal) [inline samples/nrf9160/aws_fota/src/main.c:505]: Same here

File: samples/nrf9160/aws_fota/src/main.c (hunk 2 above)
Comment (sigvartmh) [reply samples/nrf9160/aws_fota/src/main.c:504]: Added

File: samples/nrf9160/aws_fota/src/main.c (hunk 3 above)
Comment (sigvartmh) [reply samples/nrf9160/aws_fota/src/main.c:507]: Not sure if I added it like you wanted but tried :)

**Comment by sigvartmh** [comment]
@rlubos Could this go in for 1.3 it's a small change.

**Pull request merged**